src/core/castImage.ts

```diff
@@ -0,0 +1,158 @@
+import Image from './Image.js'
+import CastImageOptions from './CastImageOptions.js'
+import PixelTypes from './PixelTypes.js'
+import IntTypes from './IntTypes.js'
+import FloatTypes from './FloatTypes.js'
+
+/**
+ * Cast an image to another PixelType or ComponentType
+ *
+ * @param {Image} image - The input image
+ * @param {CastImageOptions} options - specify the componentType and/or pixelType of the output
+ */
+function castImage (inputImage: Image, options: CastImageOptions): Image {
+  const outputImageType = { ...inputImage.imageType }
+
+  if (typeof options.pixelType !== 'undefined') {
+    outputImageType.pixelType = options.pixelType
+    if (options.pixelType === PixelTypes.Scalar && outputImageType.components !== 1) {
+      throw new Error('Cannot cast multi-component image to a scalar image')
```

Comment:
Would it also make sense to check data range and warn about data loss due to narrowing conversions?

Reply:
Worth considering, but in the trade-off of developer safety nets and performance, we do not want to iterate through the data here to compute the range.